components/DownloadsComponent.tsx

```diff
@@ -0,0 +1,165 @@
+"use client"
+import React, {useState, useEffect} from 'react';
+import OperatingSystems from './OperatingSystems';
+import Architectures from './Architectures';
+import Version from './Version';
+import ReleasesTable from './ReleasesTable';
+import { Box } from '@mui/material';
+import { RotatingLines } from "react-loader-spinner";
+import allFilteredAssets, {FilteredAsset} from "../utils/fetchReleases";
+import { useTheme } from '@mui/material/styles';
+
+const DownloadsComponent: React.FC  = () => {
+    const [originalData, setOriginalData] = useState<FilteredAsset[]>([]);
+    const [displayData, setDisplayData] = useState<FilteredAsset[]>([]);
```

Comment:
Remove displayData and use useMemo to generate it after originalData is set. In general you don't want to use state for something that can be calculated from state.

Reply:
This is done.